### PR TITLE
Use jest for stylelint-polaris tests

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Converging on jest as our test runner of choice seems like a good idea.

This PR converts's stylelint-polaris's test scripts to be jest tests authored in the help of [jest-preset-stylelint](https://www.npmjs.com/package/jest-preset-stylelint)'s testRule function.

I spent a bit of the afternoon pairing with @alex-page and we came up with this.

### WHAT is this pull request doing?

Replace the test script in stylelint-polaris with tests ran by jest.
The `test` command in stylelint-polaris now runs jest.

### How to 🎩

- Note running `yarn test --filter="@shopify/stylelint-polaris" --no-cache` runs jest tests
- Note running `yarn test:watch` runs tests from stylelint-polaris too.

### CODE REVIEW

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc-",
+    },
+    {
+      code: '.a { color: --p-text; }',
+      description: 'Using custom-properties from polaris-tokens',
+    },
+    {
+      code: '.a { --test: red; color: --test; }',
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js]:
```suggestion
      code: '.a { --test: red; color: var(--test); }',
```

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      code: '.a { border: 1px red; }',
+      description: 'Uses no custom properties',
+    },
+    {
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc-",
+    },
+    {
+      code: '.a { color: --p-text; }',
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js]:
```suggestion
      code: '.a { color: var(--p-text); }',
```

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      description:
+        'Defining custom-properties that start with --pc- is disallowed',
+      message: messages.rejected('--pc-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 21,
+    },
+    {
+      code: '.a { color: --p-absent; }',
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js]:
```suggestion
      code: '.a { color: var(--p-unknown); }',
```

Included light suggestion to use `unknown` to indicate usage of a non-polaris-token

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      message: messages.rejected('--p-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 20,
+    },
+    {
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc- is disallowed",
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js:152]:
```suggestion
        "Defining custom-properties that don't start with --pc- is disallowed",
```

Removing `--p-` as those should all be generated

File: stylelint-polaris/plugins/global-disallowed-list/index.test.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+testRule({
+  ruleName,
+  plugins: [__dirname],
+  config,
+  customSyntax: 'postcss-scss',
+  accept: [
+    {
+      code: '.valid { color: red; }',
+      description: 'Uses nothing on the banned list',
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/global-disallowed-list/index.test.js]:
```suggestion
      description: 'Uses nothing on the deny list',
```
Light suggestion to use `allow|deny list` convention

File: stylelint-polaris/plugins/global-disallowed-list/index.test.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    },
+  ],
+
+  reject: [
+    {
+      code: '.invalid { color: --p-button-font; }',
+      description: 'Uses something on the banned list',
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/global-disallowed-list/index.test.js]:
```suggestion
      description: 'Uses something on the deny list',
```

**Review by aaronccasanova — APPROVED** [review]
Thanks for kicking this off and establishing a solid pattern for future `stylelint-polaris` tests! Looks like we just need to resolve some CI errors and update all custom-property usages to include `var(...)` before we can shipit 🙌

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js (hunk 4 above)
Comment (BPScott) [reply stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js:152]: This block is for the internal config. Defining prpoerties that begin with `--p` in css is disallowed (see the test directly above this one). the `--p` properties are all provided by the CustomProperties component, so you're not allowed to create them in css files.

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js (hunk 3 above)
Comment (BPScott) [reply stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js]: Updated, thanks!

File: stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js (hunk 4 above)
Comment (aaronccasanova) [reply stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js:152]: > the `--p` properties are all provided by the CustomProperties component, so you're not allowed to create them in css files.

Right 👍 However, this `description` basically states you ARE allowed to define `--p-` custom properties.
e.g. the contrapositive
```
Defining custom-properties that start with --p- or --pc- are allowed
```

Unless I'm misunderstanding (double negatives are hard). Feel free to resolve the issue.

File: stylelint-polaris/plugins/global-disallowed-list/index.test.js (hunk 5 above)
Comment (BPScott) [reply stylelint-polaris/plugins/global-disallowed-list/index.test.js]: I went with "disallowed list" to match the rule name

**Review by alex-page — APPROVED** [review]

**Pull request merged**